cmd/clusterctl/api/v1alpha3/provider_type.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha3
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +kubebuilder:resource:path=providers,scope=Namespaced,categories=cluster-api
+// +kubebuilder:storageversion
+// +kubebuilder:object:root=true
+// +kubebuilder:printcolumn:name="Type",type="string",JSONPath=".type"
+// +kubebuilder:printcolumn:name="Version",type="string",JSONPath=".version"
+// +kubebuilder:printcolumn:name="Watch Namespace",type="string",JSONPath=".watchedNamespace"
+
+// Provider is the Schema for the providers API
+type Provider struct { //nolint
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Type indicates the type of the provider.
+	// See ProviderType for a list of supported values
+	// +optional
+	Type string `json:"type,omitempty"`
+	// Version indicates the component version.
+	// +optional
+	Version string `json:"version,omitempty"`
+	// WatchedNamespace indicates the namespace where the provider controller is is watching.
+	// if empty the provider controller is watching for objects in all namespaces.
+	// +optional
+	WatchedNamespace string `json:"watchedNamespace,omitempty"`
```

Comment:
Nit: leave a space between each field and the next comment, it helps with readability

Reply:
+1